From 17eb272e0cc6dd8404d210770ab9c2c0c9624a09 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 2 May 2012 16:57:04 +0100
Subject: [PATCH] lib: Fix generic stylus missing an eraser

HasEraser=yes, but we didn't add an eraser for the stylus,
leading to bugs in the display UI for consumer products.

https://bugzilla.gnome.org/show_bug.cgi?id=675299
---
 libwacom/libwacom-database.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/libwacom/libwacom-database.c b/libwacom/libwacom-database.c
index 0c60d6d..9fc9e35 100644
--- a/libwacom/libwacom-database.c
+++ b/libwacom/libwacom-database.c
@@ -369,9 +369,10 @@ libwacom_parse_tablet_keyfile(const char *path)
 		g_strfreev (styli_list);
 		device->supported_styli = (int *) g_array_free (array, FALSE);
 	} else {
-		device->supported_styli = g_new (int, 1);
-		*device->supported_styli = 0xfffff;
-		device->num_styli = 1;
+		device->supported_styli = g_new (int, 2);
+		device->supported_styli[0] = WACOM_STYLUS_FALLBACK_ID;
+		device->supported_styli[1] = WACOM_ERASER_FALLBACK_ID;
+		device->num_styli = 2;
 	}
 
 	/* Features */
-- 
1.7.8.6

